src/rust/engine/src/rule_graph.rs

```diff
@@ -842,14 +842,34 @@ fn get_str(get: &Get) -> String {
 /// TODO: Move all of these methods to Display impls.
```

Comment:
This is an important thing to consider.

Most likely we're going to end up with different implementations of rendering per-usecase (which is fine, I think). But at least here, these are no longer generic entry rendering functions; rather, graphviz-specific functions. I'm not exactly sure how to square that, but regardless: should remove this comment if these are no longer generic.

Reply:
I totally agree! I should have called this a WIP given that there is remaining TODO. I will add this to the TODO and mark this PR as WIP.

Reply:
I believe this is the cause of the unit test failures as well.